spec/controllers/admin/auctions_controller_spec.rb

```diff
@@ -16,10 +17,40 @@
 
   describe '#show' do
     it 'assigns presented auction' do
+      user = create(:admin_user)
       auction_record = create(:auction)
       get :show, { id: auction_record.id }, user_id: user.id
       auction = assigns(:auction)
       expect(auction.auction).to eq(auction_record)
     end
   end
+
+  describe '#update' do
+    context 'when the params are invalid' do
+      it 'assigns the un-updated auction' do
+        user = create(:admin_user)
+        auction_record = create(:auction)
+
+        update_auction_double = double
+        allow(update_auction_double).to receive(:perform)
+          .and_return(false)
+        allow(UpdateAuction).to receive(:new)
+          .with(any_args)
+          .and_return(update_auction_double)
+
+        params = {
+          id: auction_record.id,
+          auction: {
+            title: 'new title'
+          }
+        }
+
+        put :update, params, user_id: user.id
+        auction = assigns(:auction)
+
+        expect(auction).to be_a(Admin::EditAuctionViewModel)
+        expect(auction.record.id).to eq(auction_record.id)
```

Comment:
thoughts on adding a third expectation:

`expect(response).to render_template(:edit)`

?